File: testSerialization.js
const { exec } = require("child_process");

const serializeData = async (data, key) => {
    const command = `./serialize_data store ${key} '${JSON.stringify(data)}'`;
    try {
        await exec(command);
        console.log(`Data serialized with key: ${key}`);
    } catch (error) {
        console.error("Serialization failed:", error);
    }
};

const generateAndSerializeData = async () => {
    const concurrency = 10; // Number of concurrent serialization tasks
    const iterations = 50;

    const promises = [];
    for (let i = 0; i < iterations; i++) {
        const data = { value: i };
        for (let j = 0; j < concurrency; j++) {
            const key = `example_key_${i}_${j}`;
            promises.push(serializeData(data, key));
        }
        if (i % 10 === 0) {
            // Log after every 10th iteration
            console.log(`Iteration ${i}: Generating and serializing data`);
        }
    }

    await Promise.all(promises);
};

generateAndSerializeData();

File: examples/test_serialization.py
import subprocess
import json, asyncio

async def serialize_data(data, key):
    command = f"./serialize_data store {key} '{json.dumps(data)}'"
    try:
        result = subprocess.run(command, shell=True, check=True)
        print(f"Data serialized with key: {key}")
    except subprocess.CalledProcessError as error:
        print("Serialization failed:", error)

async def generate_and_serialize_data():
    concurrency = 10  # Number of concurrent serialization tasks
    iterations = 50

    tasks = []
    for i in range(iterations):
        data = {"value": i}
        for j in range(concurrency):
            key = f"example_key_{i}_{j}"
            tasks.append(serialize_data(data, key))
        if i % 10 == 0:
            # Log after every 10th iteration
            print(f"Iteration {i}: Generating and serializing data")

    # Wait for all tasks to complete
    await asyncio.gather(*tasks)

asyncio.run(generate_and_serialize_data())